Kernel/FileSystem/Ext2FS/FileSystem.cpp

```diff
@@ -29,12 +29,12 @@ Ext2FS::~Ext2FS() = default;
 ErrorOr<void> Ext2FS::rename(Inode& old_parent_inode, StringView old_basename, Inode& new_parent_inode, StringView new_basename)
 {
     if (!new_parent_inode.lookup(new_basename).is_error())
-        TRY(new_parent_inode.remove_child(new_basename));
```

Comment:
I'm probably just missing something, but why do we only want to not remove the "." and ".." entries for ext2 filesystems in `rename`?

Reply:
Ah oops sorry I just got confused by the commit order. For FATFS you already added a `remove_child_impl` with a `free_clusters` argument in the commit before that.
And RAMFS doesn't need that since it generates the `.` and `..` entries on the fly.